app/models/va_facility.rb

```diff
@@ -49,29 +46,25 @@ def self.find_by(id:)
     prefix, station = id.split('_')
     adapter = client_adapter(ID_PREFIXES[prefix])
     return nil unless adapter
-    request = adapter.find_by(id: station)
-    responses = multi_client.run([request])
-    adapter.class&.from_gis(responses.first.first) unless responses.first.blank?
+    adapter.get(station)
   end
 
   def self.service_whitelist(prefix)
-    client_adapter(prefix)&.service_whitelist
-  end
-
-  def self.multi_client
-    @mc ||= Facilities::MultiClient.new
+    client_adapter(prefix)&.adapter&.service_whitelist
   end
 
   def self.client_adapter(prefix)
-    @client_map ||= init_adapters
+    @lock.synchronize do
+      @client_map ||= init_adapters
+    end
     @client_map[prefix]
   end
 
   def self.init_adapters
     {
-      HEALTH => VHAFacilityAdapter.new,
-      CEMETERY => NCAFacilityAdapter.new,
-      BENEFITS => VBAFacilityAdapter.new
+      HEALTH => Facilities::LocalStore.new(ENV['VHA_MAPSERVER_URL'], nil, VHAFacilityAdapter),
+      CEMETERY => Facilities::LocalStore.new(ENV['NCA_MAPSERVER_URL'], nil, NCAFacilityAdapter),
```

Comment:
This might be for a future refactor, but is it really necessary to have adapters? Why not just 3 separate clients? And a single client that manages the mutex synchronization?

For keeping code DRY you could use mixins for any components that are common to all three clients. I think this would be more the Ruby way.

I think this would help a little bit with readability, knowing why there is a need for 3 separate APIs, etc.